pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/leap/LeapDriver.java

```diff
@@ -96,12 +99,16 @@ public String initialState() {
         if (response.statusCode() != 200) {
             throw new RuntimeException(MessageFormat.format("HTTP server returned status code {0}", response.statusCode()));
         };
-        log.trace("pollEvents: body={}", response.body());
-        final long timestampNanos = System.currentTimeMillis() * 1000 * 1000;
-        final byte[] bytes = response.body().getBytes(StandardCharsets.UTF_8);
-        final String routingKey = getRoutingKey();
-        final PersistentQueueElement event = new PersistentQueueElement(bytes, routingKey, timestampNanos);
-        events.add(event);
+        log.trace("pollEvents: body={}", response.body());        
+        JsonNode jsonNode = mapper.readTree(response.body());        
+        final ArrayNode arrayNode = (ArrayNode) jsonNode;
+        for (JsonNode node : arrayNode) {
+            final long timestampNanos = System.currentTimeMillis() * 1000 * 1000;
```

Comment:
Calculate timestampNanos once before the loop so that all events have the same timestamp.

Reply:
Done